velox/type/Date.h

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include "velox/type/StringView.h"
+
+#include <iomanip>
+#include <sstream>
+#include <string>
+
+#include <folly/dynamic.h>
+
+namespace facebook::velox {
+
+struct Date {
+ public:
+  constexpr Date() : days_(0) {}
+  constexpr Date(int32_t days) : days_(days) {}
+
+  int32_t getDays() const {
+    return days_;
+  }
+
+  bool operator==(const Date& b) const {
+    return days_ == b.days_;
+  }
+
+  bool operator!=(const Date& b) const {
+    return days_ != b.days_;
+  }
+
+  bool operator<(const Date& b) const {
+    return days_ < b.days_;
+  }
+
+  bool operator<=(const Date& b) const {
+    return days_ <= b.days_;
+  }
+
+  bool operator>(const Date& b) const {
+    return days_ > b.days_;
+  }
+
+  bool operator>=(const Date& b) const {
+    return days_ >= b.days_;
+  }
+
+  // Needed for serialization of FlatVector<Date>
+  operator StringView() const {
+    return StringView("TODO: Implement");
+  };
+
+  std::string toString() const {
+    // Find the number of seconds for the days_;
```

Comment:
Is it important to format the date like so? Can we just return "N days since epoch"? If this function does need to do non-trivial work, then let's move it to .cpp file.

Reply:
I did research around, and that does seem like the format to represent a Date value. Although internally represented as the days since epoch, I've almost never seen it printed out that way.

Moving this code to the cpp file though.